crypto/ocsp/ocsp_asn.c

```diff
@@ -22,6 +29,30 @@ ASN1_SEQUENCE(OCSP_CERTID) = {
 
 IMPLEMENT_ASN1_FUNCTIONS(OCSP_CERTID)
 
+ASN1_SEQUENCE(OCSP_ONEREQ) = {
+    ASN1_SIMPLE(OCSP_ONEREQ, reqCert, OCSP_CERTID),
+    ASN1_EXP_SEQUENCE_OF_OPT(OCSP_ONEREQ, singleRequestExtensions,
+                             X509_EXTENSION, 0)
+} ASN1_SEQUENCE_END(OCSP_ONEREQ)
+
+IMPLEMENT_ASN1_FUNCTIONS(OCSP_ONEREQ)
+
+ASN1_SEQUENCE(OCSP_REQINFO) = {
+    ASN1_EXP_OPT(OCSP_REQINFO, version, ASN1_INTEGER, 0),
+    ASN1_EXP_OPT(OCSP_REQINFO, requestorName, GENERAL_NAME, 1),
+    ASN1_SEQUENCE_OF(OCSP_REQINFO, requestList, OCSP_ONEREQ),
+    ASN1_EXP_SEQUENCE_OF_OPT(OCSP_REQINFO, requestExtensions, X509_EXTENSION, 2)
```

Comment:
From the ASN.1 definition what do the 0 and 2 do here vs line 35?
* singleRequestExtensions    [0] EXPLICIT Extensions OPTIONAL
* requestExtensions   [2] EXPLICIT Extensions OPTIONAL

Reply:
These are context-specific ASN.1 TAGS to unambiguously identify certain components in an encoded stream. These context specific tags are used to eliminate ambiguity in a `ASN.1 SEQUENCE` which contains `OPTIONAL` elements.
"If you know you are receiving two items that are not optional, one after the other, you know which is which even if their tags are the same. However, if the first one is optional, the two must have different tags, or you would not be able to tell which one you had received if only one was present in the encoding."

* reference: https://stackoverflow.com/a/15071901

Reply:
https://luca.ntop.org/Teaching/Appunti/asn1.html is a helpful resource that I liked when working on the pkcs#8 pieces.

The `0` and `1` get encoded into the lower 6 bits of the (T)ag portion of the TLV of the DER encoding. Then as indicated earlier the explicit controls whether the underlying type's tag information is then encoded into the V portion as a nested TLV pair.